advice.c

```diff
@@ -261,3 +261,17 @@ void detach_advice(const char *new_name)
 
 	fprintf(stderr, fmt, new_name);
 }
+
+void advise_on_moving_dirty_path(struct string_list *pathspec_list)
+{
+	struct string_list_item *item;
+
+	if (!pathspec_list->nr)
+		return;
+
+	fprintf(stderr, _("The following dirty paths and/or pathspecs are moved\n"
+			  "but not sparsified. Use \"git add\" to stage them then\n"
+			  "use \"git sparse-checkout reapply\" to sparsify them.\n"));
+	for_each_string_list_item(item, pathspec_list)
+		fprintf(stderr, "%s\n", item->string);
```

Comment:
A couple of suggestions to make the advice clearer:
- Separate the warning message from the advice to fix the issue and allow the user to disable the non-warning advice with the `advice.updateSparsePath` config (see `advise_on_updating_sparse_paths()`).
- Recommend `--sparse` flag for `git add` (since the user would be adding paths outside the sparse checkout definition).
- Change "sparsify" to more precise references to "sparse-checkout definition" and/or "sparsity rules" (for consistency with `advise_on_updating_sparse_paths()`)

You don't need to use my exact wording (below) if you'd rather write something else, but at a high level these suggestions should clarify things for users.

```suggestion
	fprintf(stderr, _("The following paths have been moved outside the\n"
			  "sparse-checkout definition but are not sparse due to local\n"
			  "modifications.  then use \"git sparse-checkout reapply\" to\n"
			  "reapply sparsity rules patterns.\n"));
	for_each_string_list_item(item, pathspec_list)
		fprintf(stderr, "%s\n", item->string);
	
	advise_if_enabled(ADVICE_UPDATE_SPARSE_PATH,
			  _("To correct the sparsity of these paths, do the following:\n"
			    "* Use \"git add --sparse <paths>\" to update the index\n"
			    "* Use \"git sparse-checkout reapply\" to apply the sparsity rules"));
```

Reply:
Thanks for the comment, they are really helpful! I was thinking about adding a new advice config, but you pointed out that I can just use `ADVICE_UPDATE_SPARSE_PATH`, which is neat!